src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -577,9 +577,9 @@ def call(
             encoder_padding_mask = invert_mask(encoder_padding_mask)
 
         # embed positions
-        positions = self.embed_positions(input_ids, use_cache=use_cache)
+        positions = self.embed_positions(input_ids, use_cache=(use_cache and decoder_cached_states is not None))
```

Comment:
TFBart required some small changes to make the pt_tf tests pass. A bigger refactor for TFBart should come soon.